Enterprise-CMCS/macpro-mdct-mcr
# [MDCT-2612] -Update validation for MLR overlay page

### Description
Validations for following MLR question fields.  

Ticket:https://qmacbis.atlassian.net/jira/software/c/projects/MDCT/boards/232?modal=detail&selectedIssue=MDCT-2612
Questions 1.3, 2.3, 3.1, and 3.4 on the MLR Overlay page should have the following validation
1. No text characters (including N/A)
2. No negative numbers
3. No Zero
4. Numbers 1 OR above only (inclusive of 1) (Positive Numbers)


Ticket https://qmacbis.atlassian.net/jira/software/c/projects/MDCT/boards/232?modal=detail&selectedIssue=MDCT-2613
Question 4.6.1 has a slightly different validation scheme
* No text characters (including N/A)
* No negative numbers
* Numbers zero or above only (inclusive of 0)


### Related ticket(s)
MDCT-2612
MDCT-2613

---
### How to test
1. Log into Medical Loss Ratio (MLR) Report
2. Create MLR Submission
3. Fill out "Information for Primary Contract"
4. 'Add Program Reporting Information'
6. Enter the MLR Reporting and fill out form
7. Test the following questions above based on the requirements

If one of the fields above is invalid, the following should happen:
1.  An inline error should be displayed
2. You should be able to save the form (with incorrect input) but unable to submit it. 


* videos a little lengthy
https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/14514294/f953e724-987b-45c6-a6d0-02bb54d85d02

---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
